DummyServer/ClientApp/src/app/home/home.component.ts

```diff
@@ -0,0 +1,159 @@
+import { Component, Inject } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Subject, Observable } from "rxjs";
+import { switchMap, startWith } from "rxjs/operators";
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+})
+export class HomeComponent {
+  public builds: Build[];
+  public branches: Branch[];
+  public definitions: Definition[];
+  public users: User[];
+
+  public selectedDefinition: Definition;
+  public selectedBranch: Branch;
+  public selectedUser: User;
+
+  constructor(private http: HttpClient, @Inject('BASE_URL') private baseUrl: string) {
```

Comment:
Doing all the initialization stuff in the constructor is considered not-good practice in Angular. https://stackoverflow.com/a/35763811/868361

One would use the ngOnInit Lifecylce hook: https://angular.io/guide/lifecycle-hooks